src/flow_graph.py

```diff
@@ -765,18 +765,18 @@ def process(item: Union[Instruction, Label]) -> None:
 
 @attr.s(eq=False)
 class BaseNode(abc.ABC):
-    block: Block = attr.ib()
+    block: Block = attr.ib(repr=False)
     emit_goto: bool = attr.ib()
-    parents: List["Node"] = attr.ib(init=False, factory=list)
-    dominators: Set["Node"] = attr.ib(init=False, factory=set)
-    immediate_dominator: Optional["Node"] = attr.ib(init=False, default=None)
-    immediately_dominates: List["Node"] = attr.ib(init=False, factory=list)
-    postdominators: Set["Node"] = attr.ib(init=False, factory=set)
-    immediate_postdominator: Optional["Node"] = attr.ib(init=False, default=None)
-    immediately_postdominates: List["Node"] = attr.ib(init=False, factory=list)
+    parents: List["Node"] = attr.ib(init=False, factory=list, repr=False)
+    dominators: Set["Node"] = attr.ib(init=False, factory=set, repr=False)
+    immediate_dominator: Optional["Node"] = attr.ib(init=False, default=None, repr=False)
+    immediately_dominates: List["Node"] = attr.ib(init=False, factory=list, repr=False)
+    postdominators: Set["Node"] = attr.ib(init=False, factory=set, repr=False)
+    immediate_postdominator: Optional["Node"] = attr.ib(init=False, default=None, repr=False)
+    immediately_postdominates: List["Node"] = attr.ib(init=False, factory=list, repr=False)
     # This is only populated on the head node of the loop,
     # i.e. there is an invariant `(node.loop is None) or (node.loop.head is node)`
-    loop: Optional["NaturalLoop"] = attr.ib(init=False, default=None)
+    loop: Optional["NaturalLoop"] = attr.ib(init=False, default=None, repr=False)
```

Comment:
Hm, did you experiment with adding a `__repr__` that delegates to `__str__`? This does feel a bit error-prone (and derived classes probably need repr=False too). Alternatively you can do `repr=False` in attr.s